dive_sailthru_client/tests/test_diveSailthruClient.py

```diff
@@ -300,6 +300,28 @@ def test_raise_exception_if_error(self, mock_response, mock_error):
         self.assertEqual(str(cm.exception), 'this is the error (1234)')
         self.assertTrue(mock_response.get_error.called)
 
+    @patch('sailthru.sailthru_response.SailthruResponse')
+    @patch('sailthru.sailthru_response.SailthruResponseError')
+    def test_raise_exception_if_error_for_email_related_errors(self, mock_response, mock_error):
+        """
+        Test that an error of one of the specific user-email-related types raises a SailthruUserEmailError
+        See the list at https://getstarted.sailthru.com/developers/api-basics/responses/
+        Our definition of the user-email-related errors is in the constant client.USER_EMAIL_ERROR_CODES, used here.
+
+        :param mock_response:
+        :param mock_error:
+        :return:
+        """
+        # set up mock response so it will simulate these errors
+        mock_response.is_ok.return_value = False
+        mock_response.get_error.return_value = mock_error
+
+        for code, message in USER_EMAIL_ERROR_CODES.items():
```

Comment:
interestingly `dict.iteritems()` was removed in python3. Technically `dict.items()` doesn't work the same in python 2 (creates a list) and python 3 (creates an iterator, which lazy iterates). Since the source data is so small I consider the difference trivial and chose to use this API since it exists in both pythons.

See https://www.python.org/dev/peps/pep-0469/#migrating-to-the-common-subset-of-python-2-and-3